### DISCUSSION TITLE
What does UV letters stand for?

### DISCUSSION THREAD

**Discussion by top-master** [libuv/libuv#3283] [Q&A] [answered]
Whenever a letter means something, then that something spelled out is more human readable.

What does UV letters stand for?

I don't want to create `libuv` or `uv` folder in my `3rd-party` directory, and may call it `universal-io` instead, one reason for that is that UV stands for something already (namely, 3D model's use *UV* for textures).

**Comment by bnoordhuis** [comment]
You can see it in the logo - it's a Unicorn Velociraptor.

**Reply by mziwisky** [reply]
apparently not originally, though -- ref: https://web.archive.org/web/20150906205017/http://logs.nodejs.org/libuv/2012-09-09

```
12:42:02	<AutowiredConstru>	nobody can tell me what uv means in the name libuv
18:34:27	<isaacs>	AutowiredConstru: Technically, "libuv" means "Librarian Unicorn Velociraptor"
18:36:06	<isaacs>	AutowiredConstru: when i asked ryah what it meant, he said, "ask piscisaureus_". then i asked piscisaureus_, and he said, "ask ryah about that"
18:36:19	<piscisaureus_>	isaacs: I might disclose the true story some day
18:36:29	<piscisaureus_>	unicorn velociraptor seems good for now
```

**Reply by top-master** [reply]
See [my real reply (link)](https://github.com/libuv/libuv/discussions/3283#discussioncomment-5741907), but here as a joke, I "disclose the true story" today:
![](https://imgs.xkcd.com/comics/goto.png)

**Comment by top-master** [comment, marked as answer]
## Answer

This needs a real fix not just an answer,
see my [Latest PR](https://github.com/libuv/libuv/pull/3981), which adds to `README.md` file, an explanation like:

```
libuv's name and logo stand for "Unicorn Velociraptor", where:
    
* U or Unicorn is a reference to universal and multi-platform.
    
* V or Velociraptor is a reference to velocity and high-performance.
```

## Why PR?

Let's say I name said folder `lib-unicorn-velociraptor` or `unicorn-velociraptor`:

* `libuv`'s README.md file does not contain "Librarian Unicorn Velociraptor" phrase.
* Hence, it may confuse other people.
* And, saying "go Google-search for it" is not very polite.

Whenever a letter **or logo** means something, then that something spelled out is both more human-readable and search-engine-friendly.

## Also
No, the logo is not a good hint, because T-rex ≠ Velociraptor:

[![compare](https://user-images.githubusercontent.com/31405473/234839323-1a6a4552-e2aa-456b-99f1-48a957e81b6f.png)](https://user-images.githubusercontent.com/31405473/235522193-92011301-7f47-4949-872e-b2f567dd0804.jpg)
>Click to zoom.


Psychology: People love little dogs,
because their faces look more like a human-baby's face,
similarly, the T-rex's snout/ nose is not that long (based on its body size), hence looks better from our human-perspective.

**Comment by alekye** [comment]
Waiting for the official answer.

**Reply by top-master** [reply]
**3 years** was not enough, wait few **years** more, maybe then ;-)
